src/strings/utf8.c

```diff
@@ -448,17 +449,18 @@ MVMuint32 MVM_string_utf8_decodestream(MVMThreadContext *tc, MVMDecodeStream *ds
                     break;
                 }
                 case UTF8_REJECT: {
-                    char *waste[] = { (char *)buffer, NULL };
-                    if (bufsize >= 3) {
-                        MVMGrapheme32 a = buffer[pos - 2], b = buffer[pos - 1], c = buffer[pos];
+                    MVMint32 error_pos = pos - 1;
```

Comment:
This is the exact same code as above, so please move it to its own function. As this is an error path anyway, the call overhead will not matter and the whole decodestream function is already quite long.